packages/adapters-library/src/scripts/helpers.ts

```diff
@@ -60,6 +60,13 @@ export class Helpers {
   }: GetPositionsInput & {
     protocolTokens: Erc20Metadata[]
   }): Promise<ProtocolPosition[]> {
+    // Otherwise we might overload the node
+    if (!protocolTokenAddresses && protocolTokens.length > 1000) {
+      throw new Error(
+        'Too many tokens to fetch, protocolTokenAddresses must be provided',
+      )
+    }
+
```

Comment:
Bit of protection here....now that we can support more pools bit worries we might try and get thousands of pools